src/scales/scale.category.js

```diff
@@ -66,8 +66,12 @@ module.exports = function(Chart) {
 			// 1 is added because we need the length but we have the indexes
 			var offsetAmt = Math.max((me.maxIndex + 1 - me.minIndex - ((me.options.gridLines.offsetGridLines) ? 0 : 1)), 1);
 
-			if (value !== undefined && isNaN(index)) {
+			// If value is a data object, then index is the index in the data array,
+			// not the index of the scale. We need to change that.
+			var valueCategory = me.isHorizontal() ? value.x : value.y;
```

Comment:
You're totally right, thank you for your feedback. Addressed this on the latest commit.